brave/src/test/java/brave/secondary_sampling/TestSecondarySampler.java

```diff
@@ -71,8 +71,8 @@ public SecondarySampler forService(String serviceName) {
       if (trigger == null) return false;
 
       // When in passive mode, we only evaluate when upstream did
-      if (trigger.mode == Trigger.Mode.PASSIVE) {
-        return state.parameter("spanId") != null;
+      if (trigger.mode == Trigger.Mode.PASSIVE && state.parameter("spanId") == null) {
```

Comment:
good catch... we should change the integration test so that it fails.

do you happen to know off hand what rule would have failed due to this?

Reply:
Nope. I looked at this to validate my understanding of `PASSIVE` mode while going through the issue #15

Reply:
ok right now the tests pass, so I'd prefer not changing the logic until they don't, especially in the test tree. This former code defines passive as literally trust the upstream decision, similar to b3=1 where nodes downstream trust upstream, except here it special cases due to the possibility of skipping.

What you've done here is have passive mean also re-assess the sampling decision, which seems confused with the idea of passive. If I caused that, please point me to the sentence, so I can clear it up.

Reply:
https://github.com/openzipkin-contrib/zipkin-secondary-sampling/blob/master/brave/src/test/java/brave/secondary_sampling/TestSecondarySampler.java#L26

How about the following instead.
`/** Triggers automatically when upstream sampled. */`

https://github.com/openzipkin-contrib/zipkin-secondary-sampling/blob/master/brave/src/test/java/brave/secondary_sampling/TestSecondarySampler.java#L73

How about the following instead.
`// When in passive mode, we always trigger when upstream did`